cluster/v1alpha1/helpers.go

```diff
@@ -293,6 +293,9 @@ func progressivePerCluster(
 
 		// If there was a breach of MaxFailures, only handle clusters that have already had workload applied
 		if !failureBreach || failureBreach && status.Status != ToApply {
+			// The length of `rolloutClusters` will be compared with the target rollout size to determine whether to return or not.
+			// The `timeoutClusters` only records clusters that are timeout and is not used for comparison with the target rollout size.
+			// In other words, the timeouted clusters are skipped and continue rollout on other clusters.
```

Comment:
depend on whether failure exceeds maxfailures

Reply:
have updated the comments.